bundles/core/org.eclipse.smarthome.core.test/src/test/java/org/eclipse/smarthome/core/net/CidrTest.java

```diff
@@ -0,0 +1,57 @@
+/**
+ * Copyright (c) 2014-2017 by the respective copyright holders.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.eclipse.smarthome.core.net;
+
+import static org.junit.Assert.assertTrue;
+
+import org.junit.Test;
+
+/**
+ * Tests for NetUtil class
+ *
+ * @author Stefan Triller - initial contribution
+ * @author David Graeff - Port to CidrAddress
+ */
+public class CidrTest {
+
+    @Test(expected = IllegalArgumentException.class)
+    public void invalidPrefix() {
+        new CidrAddress("192.168.5.8/-1");
+    }
+
+    @Test(expected = IllegalArgumentException.class)
+    public void outOfBoundsPrefix() {
+        new CidrAddress("192.168.5.8/33");
```

Comment:
Does this work for IPv6 addresses? (i.e. is "2001:db8:0:0:8:800:200c:417a/33" valid)?

Reply:
I have added test cases for IPv6